examples/core/rgcn/hetero_rgcn.py

```diff
@@ -0,0 +1,647 @@
+        }
+
+        # Create a dictionary of node features for the destination nodes in
+        # the graph. We slice the node features according to the number of
+        # destination nodes of each type. This is necessary because when
+        # incorporating the effect of self-loop edges, we perform computations
+        # only on the destination nodes' features. By doing so, we ensure the
+        # feature dimensions match and prevent any misuse of incorrect node
+        # features.
+        inputs_dst = {
+            k: v[: g.number_of_dst_nodes(k)] for k, v in inputs.items()
+        }
+
+        # Apply the convolution operation on the graph. mod_kwargs are
+        # additional arguments for each relation function defined in the
+        # HeteroGraphConv. In this case, it's the weights for each relation.
+        hs = self.conv(g, inputs, mod_kwargs=wdict)
+
+        def _apply(ntype, h):
+            h = h + self.loop_weights[ntype](inputs_dst[ntype])
+            if self.activation:
+                h = self.activation(h)
+            return self.dropout(h)
+
+        # Apply the function defined above for each node type. This will update
+        # the node features using the self-loop weights, apply the activation
+        # function and dropout.
+        return {ntype: _apply(ntype, h) for ntype, h in hs.items()}
+
+
+class EntityClassify(nn.Module):
+    def __init__(self, g, in_dim, out_dim):
+        super(EntityClassify, self).__init__()
+        self.in_dim = in_dim
+        self.h_dim = 64
+        self.out_dim = out_dim
+
+        # Generate and sort a list of unique edge types from the input graph.
+        # eg. ['writes', 'cites']
+        self.rel_names = list(set(g.etypes))
+        self.rel_names.sort()
+        self.dropout = 0.5
+
+        self.layers = nn.ModuleList()
+
+        # First layer: transform input features to hidden features. Use ReLU
+        # as the activation function and apply dropout for regularization.
+        self.layers.append(
+            RelGraphConvLayer(
+                self.in_dim,
+                self.h_dim,
+                g.ntypes,
+                self.rel_names,
+                activation=F.relu,
+                dropout=self.dropout,
+            )
+        )
+
+        # Second layer: transform hidden features to output features. No
+        # activation function is applied at this stage.
+        self.layers.append(
+            RelGraphConvLayer(
+                self.h_dim,
+                self.out_dim,
+                g.ntypes,
+                self.rel_names,
+                activation=None,
+            )
+        )
+
+    def reset_parameters(self):
+        # Reset the parameters of each layer.
+        for layer in self.layers:
+            layer.reset_parameters()
+
+    def forward(self, h, blocks):
+        for layer, block in zip(self.layers, blocks):
+            h = layer(block, h)
+        return h
+
+
+class Logger(object):
```

Comment:
Instead of copy-paste from PyG, can we design a logger to make the output format of all our examples consistent, not just for hetero_rgcn.

Reply:
Get, I'll follow this issue: **Design a Logger class for `core` examples.**